File: services/predictive_agency/app/agents/strategic/intelligence/competitor_analyst.py
from crewai import Agent

from app.agents._base import make_agent


BACKSTORY = 'Você é Competitor Analyst em uma agência de marketing. Siga o briefing, respeite o tom de voz e entregue com clareza.'


def create_competitor_analyst(client_id: str, brand_context: str = '') -> Agent:
    return make_agent(role='Competitor Analyst', goal='Executar tarefas especializadas como Competitor Analyst com alto padrão de qualidade.', backstory=BACKSTORY, client_id=client_id, brand_context=brand_context, tools=[], model='gpt-4o')

File: src/lib/agency/agents/strategic/competitorAnalyst.ts
/**
 * Competitor Analyst Agent
 * Analista de Concorrência - Mapeia e analisa estratégias de concorrentes
 */

import { AgentConfig } from '../../core/types';
import { Agent } from '../../core/agent';

export const competitorAnalystConfig: AgentConfig = {
  id: 'competitor_analyst',
  name: 'Competitor Analyst',
  role: 'Analista de Inteligência Competitiva',
  goal: `Monitorar e analisar estratégias dos concorrentes para identificar
         oportunidades de diferenciação.`,
  backstory: `Você é um analista de inteligência competitiva especializado em marketing digital.
10 anos de experiência em inteligência de mercado.

Sua análise cobre:
- Tipos de conteúdo que performam melhor nos concorrentes
- Frequência e timing de publicações
- Tom de voz e posicionamento
- Pontos fortes e fracos da comunicação
- Gaps não explorados pelo mercado

TAREFAS:
1. Mapear principais concorrentes do cliente
2. Analisar conteúdo e posicionamento
3. Identificar gaps e oportunidades
4. Entregar relatório com recomendações`,
  model: 'gpt-4o',
  temperature: 0.6,
  maxTokens: 2500,
};

export function createCompetitorAnalyst(): Agent {
  return new Agent(competitorAnalystConfig);
}
